src/encoded/types/biosample.py

```diff
@@ -709,6 +717,13 @@ def generate_summary_dictionary(
             pluralize(post_treatment_time, post_treatment_time_units)
             )
 
+    if post_nucleic_acid_delivery_time is not None and \
+        post_nucleic_acid_delivery_time_units is not None:
+        dict_of_phrases['post_nucleic_acid_delivery_time'] = '{} post-nucleic acid delivery time'.format(
```

Comment:
As-is this is correct and is a good way to do it. If interested, since we updated to Python 3.6+, you can use f-string (https://cito.github.io/blog/f-strings/). I feel it is an even better way.